File: app/src/main/java/pl/pola_app/ui/fragment/KeyboardFragment.java
package pl.pola_app.ui.fragment;

import android.app.Activity;
import android.app.Fragment;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import pl.pola_app.ui.activity.MainActivity;
import pl.pola_app.ui.keyboard.KeyboardListener;
import pl.pola_app.ui.keyboard.NumericKeyboard;

public class KeyboardFragment extends Fragment implements KeyboardListener {

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        final NumericKeyboard numericKeyboard = new NumericKeyboard(inflater.getContext());
        final ViewGroup.LayoutParams layoutParams = new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
        numericKeyboard.setLayoutParams(layoutParams);
        numericKeyboard.setKeyboardListener(this);

        return numericKeyboard;
    }

    @Override
    public void onInputConfirmed(@NonNull String text) {
        final Activity activity = getActivity();
        ((MainActivity) activity).onBarcode(text, false);
        activity.getFragmentManager().popBackStack();
    }
}

File: app/src/main/java/pl/pola_app/ui/fragment/KeyboardFragment.kt
package pl.pola_app.ui.fragment

import android.app.Fragment
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import pl.pola_app.ui.activity.MainActivity
import pl.pola_app.ui.keyboard.KeyboardListener
import pl.pola_app.ui.keyboard.NumericKeyboard

class KeyboardFragment : Fragment(), KeyboardListener {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        val numericKeyboard = NumericKeyboard(inflater.context)
        val layoutParams = ViewGroup.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT,
            ViewGroup.LayoutParams.MATCH_PARENT
        )
        numericKeyboard.layoutParams = layoutParams
        numericKeyboard.setKeyboardListener(this)
        return numericKeyboard
    }

    override fun onInputConfirmed(text: String) {
        val activity = activity
        (activity as MainActivity).onBarcode(text, false)
        activity.getFragmentManager().popBackStack()
    }
}